File: 2748-number-of-beautiful-pairs/2748-number-of-beautiful-pairs.py
class Solution:
    def countBeautifulPairs(self, nums):
        return sum(
            sum(1 for j in range(i + 1, len(nums)) if gcd(int(str(nums[i])[0]), nums[j] % 10) == 1)
            for i in range(len(nums))
        )

File: 2748-number-of-beautiful-pairs/2748-number-of-beautiful-pairs.rb
# @param {Integer[]} nums
# @return {Integer}
def count_beautiful_pairs(nums)
  nums.each_index.sum do |i|
    (i+1...nums.size).count do |j|
      nums[i].to_s[0].to_i.gcd(nums[j] % 10) == 1
    end
  end
end